### PR TITLE
Version-specific rotary haptics

### PR DESCRIPTION
**Pull request by Kpeved**
#### WHAT
Rotary Haptics were changed to rely on Wear version rather than specific devices. That should help to bring consistency across different devices with Rotary. 

#### WHY
Vibrate "Using Crown" setting wasn't properly working on devices other than Pixel Watch (https://github.com/google/horologist/issues/1750 ) - 

#### HOW
A separate version specific logic was added.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

**Review by yschimke — APPROVED** [review]

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 1]
```diff
@@ -262,19 +265,21 @@ public fun rememberDefaultRotaryHapticFeedback(): RotaryHapticFeedback =
...
-        DefaultRotaryHapticFeedback(view)
+        PreWear3point5RotaryHapticFeedback(view)
     }
 
 /**
  * Default Rotary implementation for [RotaryHapticFeedback]
  */
 @ExperimentalHorologistApi
-public class DefaultRotaryHapticFeedback(private val view: View) : RotaryHapticFeedback {
+public class PreWear3point5RotaryHapticFeedback(private val view: View) : RotaryHapticFeedback {
```
[the remark this answers is not in the document]
Comment (Kpeved) [reply compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt]: No need, changed to private

**Comment by yschimke** [comment]
Failing on NPE here

```
internal fun getWearPlatformMrNumber(context: Context): Int =
    Settings.Global
        .getString(context.contentResolver, WEAR_PLATFORM_MR_NUMBER).toIntOrNull() ?: 0
```

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 2]
```diff
@@ -374,5 +398,15 @@ private class GalaxyWatchClassicHapticFeedback(private val view: View) : RotaryH
...
+private fun isWear3point5(context: Context): Boolean =
+    Build.VERSION.SDK_INT == 30 && getWearPlatformMrNumber(context) >= 5
+
+@ChecksSdkIntAtLeast(api = Build.VERSION_CODES.TIRAMISU)
+private fun isWear4AtLeast(): Boolean =
+    Build.VERSION.SDK_INT >= 33
+
+internal fun getWearPlatformMrNumber(context: Context): Int =
+    Settings.Global
+        .getString(context.contentResolver, WEAR_PLATFORM_MR_NUMBER).toIntOrNull() ?: 0
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:410]:
```suggestion
        .getString(context.contentResolver, WEAR_PLATFORM_MR_NUMBER)?.toIntOrNull() ?: 0
```

File: compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt [hunk 3]
```diff
@@ -371,8 +397,21 @@ private class GalaxyWatchClassicHapticFeedback(private val view: View) : RotaryH
     }
 }
 
+private fun isGalaxyWatch(): Boolean =
+    Build.MANUFACTURER.contains("Samsung", true)
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/rotaryinput/Haptics.kt:401]:
```suggestion
    Build.MANUFACTURER.contains("Samsung", ignoreCase = true)
```

**Pull request merged**